Program.cs

```diff
@@ -0,0 +1,129 @@
+﻿namespace Homework7
+{
+    internal class Program
+    {
+        static void Main(string[] args)
+        {
+            //Car MyCar = new Car("Ford", "Mustang", 1995, 23156.0f);
+
+            //Console.WriteLine("Drive() Method");
+            //MyCar.Drive(123.35f);
+            //Console.WriteLine(MyCar.WriteOutInfo("mileage"));
+
+            //Console.WriteLine("\n");
+            //Console.WriteLine("DisplayCarInfo() Method");
+            //MyCar.DisplayCarInfo();
+
+            //Console.WriteLine("\n");
+            //Console.WriteLine(MyCar);
+            //Console.WriteLine("Press any key...");
+            //Console.ReadKey();
+
+
+            //Book RandomBook = new Book("The Great Gatsby", "F. Scott Fitzgerald", 180);
+            //RandomBook.Read(50);
+            //RandomBook.DisplayProgress();
+            //Console.WriteLine("Press any key...");
+            //Console.ReadKey();
+
+
+        }
+    }
+
+    public class Car
+    {
+        private string Name { get; } //Pořád si nejsem jistý, kdy je dobrá praxe používat private-public,
+        private string Model { get; } //ale říkal jsem si, že kdybych měl někde v databázi uvedené auté, 
+        private int Year { get; } //tak bych ho nechtěl jen tak změnit. Zvlášť mileage kvůli scamům.
+        private float Mileage { get; set; }
+
+        public Car(string name, string model, int year, float mileage)
+        {
+            Name = name;
+            Model = model;
+            Year = year;
+            Mileage = mileage;
+        }
+
+        public void Drive(float kiloMeters)
+        {
+            if (kiloMeters < 0)
+            {
+                throw new ArgumentException("Kilometers driven cannot be negative.");
+            }
+            this.Mileage = this.Mileage + kiloMeters;
+        }
+        public string WriteOutInfo(string input)
+        {
+            if (input == "name" || input == "Name" || input == "NAME")
```

Comment:
Tohle je trochu tricky, protoze pak bys musel vypsat vsechny kombinace, kterych muze byt nekonecno (napr. NaMe apod mixy malych a velkych pismen).

Hezky by se tenhle typ ifu dal napsat takhle

```suggestion
            if (string.Equals(input, nameof(this.Name), StringComparison.OrdinalIgnoreCase))
```